exporter/elasticsearchexporter/config.go

```diff
@@ -108,6 +111,32 @@ func (c *BatcherConfig) Unmarshal(conf *confmap.Conf) error {
 	return nil
 }
 
+type IncludeSourceOnErrorSettings struct {
+	// Enabled configures whether the bulk index responses include
+	// a part of the source document on error.
+	// Defaults to false.
+	//
+	// This setting requires Elasticsearch 8.18+. Using it in prior versions
+	// have no effect.
+	// This setting is overridden by CompatibilityMode if CompatibilityMode
+	// is true.
+	//
+	// WARNING: if set to true, the user is responsible for sanitizing the
+	// error as it may contain sensitive data from the source.
+	Enabled bool `mapstructure:"enabled"`
+
+	// CompatibilityMode exists as a backward compatible option for older
+	// versions of Elasticsearch, to avoid logging sensitive data.
+	// Defaults to true.
+	//
+	// If set to true, the bulk indexer discards the error reason from the
+	// bulk index responses entirely.
+	//
+	// This setting has a higher precedence than Enabled, i.e. if this is
+	// true, Enabled will be ignored.
+	CompatibilityMode *bool `mapstructure:"compatibility_mode"`
```

Comment:
It can be just a bool. Change the default config in createDefaultConfig to make it true by default.